spec/system/job_applications/status_transition_spec.rb

```diff
@@ -0,0 +1,365 @@
+require "rails_helper"
+
+RSpec.describe "check job application after status transition" do
+  let(:publisher) { create(:publisher, :with_organisation, accepted_terms_at: 1.day.ago) }
+  let(:organisations) { publisher.organisations }
+  let(:vacancy) { create(:vacancy, organisations:) }
+  let(:jobseeker) { create(:jobseeker) }
+  let(:job_application) { jobseeker.reload.job_applications.where(status:).first }
+  let(:status) { "" }
+
+  before do
+    JobApplication.statuses.each_key do |status|
+      create(:job_application, :"status_#{status}", jobseeker:)
+    end
+
+    if job_application.present?
+      job_application.vacancy = vacancy
+      job_application.save!
+    end
+  end
+
+  describe "jobseeker job applications listing" do
+    it "jobseeker can view all its applications" do
+      run_with_jobseeker(jobseeker) do
+        jobseeker_applications_page.load
+
+        jobseeker.job_applications.each do |ja|
+          tag_text = JobApplicationsHelper::JOBSEEKER_STATUS_MAPPINGS[ja.status.to_sym]
+          application = jobseeker_applications_page.job_application(ja.id)
+          expect(application).to be_present, "missing job application status #{ja.status}"
+          expect(application.tag).to have_text(tag_text)
+        end
+      end
+    end
+  end
+
+  describe "transition: no job application to draft" do
+    it "jobseeker can start a job application" do
+      run_with_jobseeker(jobseeker) do
+        #
+        # jobseeker starts an application
+        #
+        jobseeker_application_start_page.load(vacancy_id: vacancy.id)
+        jobseeker_application_start_page.btn_start_application.click
+        jobseeker.reload
```

Comment:
Would be worth to assert on the `jobseeker.job_applications.draft.count` increasing by 1 on this click?

This or any other way to check that when visiting the Jobseeker Applications page there is a new draft there.